### PR TITLE
feat: Remove unneeded operator flags

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request (what was done and why):

- Remove CLI unused CLI flags and converted them to constants
- Cleanup LogPipeline validation since the forbidden plugin are no longer configurable
- Fix sample

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/1122

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

File: apis/telemetry/v1beta1/logpipeline_validation_test.go [hunk 1]
```diff
@@ -182,8 +152,7 @@ func TestValidateCustomFiltersContainsNoName(t *testing.T) {
 		},
 	}
 
-	vc := getLogPipelineValidationConfig()
-	err := logPipeline.validateFilters(vc.DeniedFilterPlugins)
+	err := logPipeline.validateFilters()
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "configuration section must have name attribute")
```
Comment (k15r) [inline apis/telemetry/v1beta1/logpipeline_validation_test.go:157]: nit (not part to be implemented as part of this PR):
lets use errors.Is to match the error and not string matching

File: apis/telemetry/v1beta1/logpipeline_validation_test.go [hunk 2]
```diff
@@ -123,8 +100,7 @@ func TestBothValueAndValueFromPresent(t *testing.T) {
 				},
 			},
 		}}
-	vc := getLogPipelineValidationConfig()
-	err := logPipeline.validateOutput(vc.DeniedOutPutPlugins)
+	err := logPipeline.validateOutput()
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "http output host must have either a value or secret key reference")
```
Comment (k15r) [inline apis/telemetry/v1beta1/logpipeline_validation_test.go:105]: nit (not part to be implemented as part of this PR):
lets use errors.Is to match the error and not string matching

File: config/samples/telemetry_v1alpha1_metricpipeline.yaml [hunk 3]
```diff
@@ -97,7 +97,7 @@ spec:
         app: metric-gen
     spec:
       containers:
-        - image: ghcr.io/skhalash/examples/monitoring-custom-metrics:3d41736
+        - image: europe-docker.pkg.dev/kyma-project/prod/examples/monitoring-custom-metrics:v20230905-b823fd14
```
Comment (k15r) [inline config/samples/telemetry_v1alpha1_metricpipeline.yaml:100]: 😆

File: main.go [hunk 4]
```diff
@@ -246,63 +192,30 @@ func main() {
...
-	flag.StringVar(&fluentBitMemoryRequest, "fluent-bit-memory-request", "50Mi", "Memory request for fluent-bit")
-	flag.StringVar(&fluentBitImage, "fluent-bit-image", defaultFluentBitImage, "Image for fluent-bit")
 	flag.StringVar(&fluentBitExporterImage, "fluent-bit-exporter-image", defaultFluentBitExporterImage, "Image for exporting fluent bit filesystem usage")
-	flag.StringVar(&fluentBitPriorityClassName, "fluent-bit-priority-class-name", "", "Name of the priority class of fluent bit ")
-	flag.IntVar(&maxLogPipelines, "fluent-bit-max-pipelines", defaultMaxNumberOfLogPipelines, "Maximum number of LogPipelines to be created. If 0, no limit is applied.")
-
+	flag.StringVar(&fluentBitImage, "fluent-bit-image", defaultFluentBitImage, "Image for fluent-bit")
+	flag.StringVar(&highPriorityClassName, "high-priority-class-name", "", "High priority class name used by managed DaemonSets")
+	flag.StringVar(&normalPriorityClassName, "normal-priority-class-name", "", "Normal priority class name used by managed Deployments")
+	flag.StringVar(&otelCollectorImage, "otel-collector-image", defaultOTelCollectorImage, "Image for OpenTelemetry Collector")
```
Comment (k15r) [inline main.go:201]: nit: please order 
- images
- priority classes

File: apis/telemetry/v1beta1/logpipeline_validation_test.go (hunk 1 above)
Comment (skhalash) [reply apis/telemetry/v1beta1/logpipeline_validation_test.go:157]: Agree, will do it in a separate PR.

File: apis/telemetry/v1beta1/logpipeline_validation_test.go (hunk 2 above)
Comment (skhalash) [reply apis/telemetry/v1beta1/logpipeline_validation_test.go:105]: Agree, will do it in a separate PR.

File: main.go [hunk 5]
```diff
@@ -72,76 +65,37 @@ import (
...
-	metricGatewayDynamicMemoryLimit   string
-	metricGatewayCPURequest           string
-	metricGatewayDynamicCPURequest    string
-	metricGatewayMemoryRequest        string
-	metricGatewayDynamicMemoryRequest string
-
-	selfMonitorImage         string
-	selfMonitorPriorityClass string
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup").WithValues("version", version)
```
Comment (k15r) [inline main.go:69]: logging the version with every logline of this logger seem a bit excessive

File: main.go [hunk 6]
```diff
@@ -246,63 +192,30 @@ func main() {
...
-	flag.StringVar(&selfMonitorPriorityClass, "self-monitor-priority-class", "", "Priority class name for self-monitor")
-
-	flag.BoolVar(&enableV1Beta1LogPipelines, "enable-v1beta1-log-pipelines", false, "Enable v1beta1 log pipelines CRD")
 
 	flag.Parse()
 
-	if err := validateFlags(); err != nil {
-		return fmt.Errorf("invalid flag provided: %w", err)
-	}
+	telemetryNamespace, _ = os.LookupEnv(telemetryNamespaceEnvVar)
```
Comment (k15r) [inline main.go:206]: slightly inconvenient not to have default values anymore. especially for local development. Now you need to run
`MANAGER_NAMESPACE=default make run`

Comment (skhalash) [reply main.go:206]: `telemetryNamespace` variable has still a default value, nothing really changed here.

File: main.go (hunk 5 above)
Comment (skhalash) [reply main.go:69]: Before, the version was added as a key to a random log entry. I can also log it once somewhere at the top of the `run` function:
```go
logger.Info("Starting Telemetry Manager", "version", version)
```

File: main.go (hunk 6 above)
Comment (skhalash) [reply main.go:206]: Ah, no, sorry, it will be overridden by `os.LookupEnv`, will fix it

Comment (skhalash) [reply main.go]: Done

File: main.go (hunk 5 above)
Comment (skhalash) [reply main.go]: Done

File: main.go (hunk 4 above)
Comment (skhalash) [reply main.go]: Done

**Review by k15r — APPROVED** [review]

**Pull request merged**